### PR TITLE
refactor validation.go to avoid duplicating

### PR DESCRIPTION
**Pull request by mqliang**
This PR helps refactor `validation.go` and expose some helpful functions and constants to `extension/validation.go`, instead of duplicating.

### CODE REVIEW

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -234,6 +235,14 @@ func ValidatePositiveQuantity(value resource.Quantity, fieldName string) errs.Va
 	return allErrs
 }
 
+func ValidateImmutableField(old, new interface{}, fieldName string) errs.ValidationErrorList {
```
Comment (brendandburns) [inline pkg/api/validation/validation.go]: Since this can only return a single error, please have it just return `error` not `errs.ValidationErrorList`

**Comment by brendandburns** [comment]
thanks for the refactor.  one comment.

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (mqliang) [inline pkg/api/validation/validation.go]: @brendandburns 
From my perspective, returning `errs.ValidationErrorList` is a better choice, just as `ValidatePositiveField` and `ValidatePositiveQuantity` do. Those three validation functions can indeed only return a single error, but there are lots of validation functions such as `ValidateLabelName` will return a list of error, keep all validation functions consistently return `errs.ValidationErrorList｀could make all invocation statements consistent.

**Comment by brendandburns** [comment]
I'm convinced by the fact that you can use `...` on an empty list without having to do the nil check...

Needs rebase.

**Comment by brendandburns** [comment]
@k8s-bot ok to test

**Pull request merged**